R/reportEnergyInvestment.R

```diff
@@ -52,21 +56,37 @@ reportEnergyInvestment <- function(gdx, regionSubsetList = NULL,
   grid  <- readGDX(gdx, name = c("teGrid", "grid"), format = "first_found")
   pebio  <- readGDX(gdx, c("peBio", "pebio"), format = "first_found")
   ttot        <- readGDX(gdx, name = "ttot", format = "first_found")
+
   # read variables
+
+  # investment cost per technology
   v_directteinv <- readGDX(gdx, name = c("v_costInvTeDir", "vm_costInvTeDir", "v_directteinv"),
                            field = "l", format = "first_found")
+  # adjustment cost per technology
   v_adjustteinv <- readGDX(gdx, name = c("v_costInvTeAdj", "vm_costInvTeAdj", "v_adjustteinv"),
                            field = "l", format = "first_found")
 
   # read parameters for additional calculations:
   pm_data   <- readGDX(gdx, c("pm_data"), format = "first_found")
 
   # data preparation
+
   ttot <- as.numeric(as.vector(readGDX(gdx, "ttot", format = "first_found")))
-  v_directteinv <- v_directteinv[, ttot, ]
-  v_adjustteinv <- v_adjustteinv[, ttot, ]
-  costRatioTdelt2Tdels <- pm_data[, , "inco0.tdelt"] / pm_data[, , "inco0.tdels"]
 
+  if (!is.null(gdx_ref)) {
```

Comment:
I would add a comment here to explain the next lines, something like: 

apply "modifyInvestmentVariables" to shift from the model-internal time coverage (deltacap and investment variables for step t represent the average of the years from t-4years to t) to the general convention for the reporting template (all variables represent the average of the years from   t-2.5years to t+2.5years)